### ISSUE TITLE
Improve Exception Handling in pline.Pileline.create()

### ISSUE THREAD

**Issue by aabdullah-bos** [amancevice/pline#3]
In the earlier versions of pline, that used boto instead of boto3 the usage model for the pline.Pipeline class was as follows:

``` python
import pline
import boto

pipeline = pline.Pipeline('SomeName', 'SomeUniquieID', 'SomeDescription')
try:
    pipeline.connect(aws_access_key_id='SomeAccessKey', aws_secret_access_key='SomeSecretKey')
except except boto.exception.NoAuthHandlerFound, err:
    # exit with some message about credentials

pipeline.create()
```

In boto3, you can connect to a resource before it is created, and therefore you no longer get the **boto.exception.NoAuthHandlerFound** exception. I would also contend that users shouldn't need to import boto in order to use pline.  I would like to request functionality (that I will probably implement myself) that allows users to follow the following pattern:

``` python
import pline
pipeline = pline.Pipeline('SomeName', 'SomeUniquieID', 'SomeDescription')
pipeline.connect(aws_access_key_id='SomeAccessKey', aws_secret_access_key='SomeSecretKey')
pipline.create()
```

IMHO, the user of a package should not have to import another package to handle an exception. The desired behavior in this case is that, we catch the botocore.exceptions.ClientError that is raised, when the credentials are not kosher.

```
pipeline.create()
---------------------------------------------------------------------------
ClientError                               Traceback (most recent call last)
<ipython-input-12-fef30f71049f> in <module>()
----> 1 pipeline.create()

/Users/aquilabdullah/.virtualenvs/web_env/lib/python2.7/site-packages/pline/pipeline.pyc in create(self)
     78             name        = self.name,
     79             uniqueId    = self.unique_id,
---> 80             description = self.desc)
     81         self.pipeline_id = response['pipelineId']
     82         if any(self.objects) or any(self.parameters):

/Users/aquilabdullah/.virtualenvs/web_env/lib/python2.7/site-packages/botocore/client.pyc in _api_call(self, *args, **kwargs)
    299                     "%s() only accepts keyword arguments." % py_operation_name)
    300             # The "self" in this scope is referring to the BaseClient.
--> 301             return self._make_api_call(operation_name, kwargs)
    302
    303         _api_call.__name__ = str(py_operation_name)

/Users/aquilabdullah/.virtualenvs/web_env/lib/python2.7/site-packages/botocore/client.pyc in _make_api_call(self, operation_name, api_params)
    396
    397         if http.status_code >= 300:
--> 398             raise ClientError(parsed_response, operation_name)
    399         else:
    400             return parsed_response

ClientError: An error occurred (UnrecognizedClientException) when calling the CreatePipeline operation: The security token included in the request is invalid.
```

**Comment by amancevice** [comment]
Sounds like a boog

**Comment by aabdullah-bos** [comment]
Thanks for the fix. Fixed in 0.4.2.
